plugin/interface_generator/main.go

```diff
@@ -378,6 +466,62 @@ func generateGlue(info *PluginInterfaceInfo) {
 	}
 }
 
+func generatePluginMetricsWrapper(info *PluginInterfaceInfo) {
+	templateFunctions := map[string]interface{}{
+		"funcStyle":   func(fields *ast.FieldList) string { return FieldListToFuncList(fields, info.FileSet) },
+		"structStyle": func(fields *ast.FieldList) string { return FieldListToStructList(fields, info.FileSet) },
+		"valuesOnly":  func(fields *ast.FieldList) string { return FieldListToNames(fields, info.FileSet, true) },
+		"destruct": func(structPrefix string, fields *ast.FieldList) string {
+			return FieldListDestruct(structPrefix, fields, info.FileSet)
+		},
+	}
+
+	// Prepare template params
+	templateParams := HooksTemplateParams{}
+	for _, hook := range info.Hooks {
+		templateParams.HooksMethods = append(templateParams.HooksMethods, MethodParams{
+			Name:   hook.FuncName,
+			Params: hook.Args,
+			Return: hook.Results,
+		})
+	}
+	for _, api := range info.API {
+		templateParams.APIMethods = append(templateParams.APIMethods, MethodParams{
+			Name:   api.FuncName,
+			Params: api.Args,
+			Return: api.Results,
+		})
+	}
+
+	pluginTemplates := map[string]string{
+		"api_metrics_generated.go":   apiMetricsWrapperTemplate,
+		"hooks_metrics_generated.go": hooksMetricsWrapperTemplate,
+	}
+
+	for fileName, presetTemplate := range pluginTemplates {
+		parsedTemplate, err := template.New("hooks").Funcs(templateFunctions).Parse(presetTemplate)
+		if err != nil {
+			panic(err)
+		}
+
+		templateResult := &bytes.Buffer{}
+		parsedTemplate.Execute(templateResult, &templateParams)
```

Comment:
Sorry for drive-by review, but this is better handled by using goimports as a library, like this:
```go
import "golang.org/x/tools/imports"

    formatted, err := imports.Process(actualFileName, sourceCode.Bytes(), &imports.Options{})
	if err != nil {
		log.Panic(err)
	}

```

Reply:
was using existing code, but this is much cleaner! great suggestion 👍